.circleci/config.yml

```diff
@@ -108,13 +117,28 @@ commands:
                 command: circleci-agent step halt
 
 jobs:
-  'Compute Merge Commit':
+  'Initialize Git':
     executor:
       name: amphtml-small-executor
     steps:
+      - restore_cache:
+          keys:
+            - source-v1-{{ .Branch }}-{{ .Revision }}
+            - source-v1-{{ .Branch }}-
+            - source-v1-
+      - checkout
+      - run:
+          name: 'Compute and Fetch Merge Commit'
+          command: |
+            ./.circleci/compute_merge_commit.sh
+            ./.circleci/fetch_merge_commit.sh
+      - save_cache:
+          key: source-v1-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - .git
       - run:
-          name: 'Compute Merge Commit'
-          command: curl -sS https://raw.githubusercontent.com/ampproject/amphtml/main/.circleci/compute_merge_commit.sh | bash
+          name: 'Store maybe_gracefully_halt.sh in Workspace'
```

Comment:
nit: For clarity, maybe call this step "Initialize Workspace"? That way, the name can stay the same even if more commands are added to it.

Reply:
Done.